markdownlint.sh

```diff
@@ -1,64 +1,113 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
 
-##### Extract some skill configuration from the incoming event payload
-fix=$( cat $ATOMIST_PAYLOAD |
-  jq -r '.skill.configuration.instances[0].parameters[] | select( .name == "push_strategy" ) | .value' )
-config=$( cat $ATOMIST_PAYLOAD |
-  jq -r '.skill.configuration.instances[0].parameters[] | select( .name == "config" ) | .value' )
-ignores=$( cat $ATOMIST_PAYLOAD |
-  jq -r '.skill.configuration.instances[0].parameters[] | select( .name == "ignores" ) | .value | join(" --ignore ")' )
-labels=$( cat $ATOMIST_PAYLOAD |
-  jq -r '.skill.configuration.instances[0].parameters[] | select( .name == "labels" ) | .value' )
-branch=$( cat $ATOMIST_PAYLOAD |
-  jq -r '.data.Push[0].branch' )
+declare Pkg=markdownlint
+declare Version=0.1.0
 
-##### Bail out early if it on a markdownlint branch
-if [[ "$branch" =~ ^markdownlint-.* ]]; then
-    exit 0
-fi
+set -o pipefail
 
-##### Make the problem matcher available to the runtime
-cp /app/markdownlint.matcher.json /atm/output/matchers/
+# print message to stdout prefixed by package name.
+# usage: msg MESSAGE
+function msg () {
+    echo "$Pkg: $*"
+}
 
-##### Create push instructions for the runtime to indicate how changes to the repo should get persisted
-if [[ -z "$labels" ]]; then
-  labels="[]"
-fi
-push=$( jq -n \
-    --arg s "$fix" \
-    --argjson l "$labels" \
-    '{
-        strategy: $s,
-        pullRequest: {
-          title: "MarkdownLint fixes",
-          body: "MarkdownLint fixed warnings and/or errors",
-          branchPrefix: "markdownlint",
-          labels: $l,
-          close: {
-            stale: true,
-            message: "Closing pull request because all fixable warnings and/or errors have been fixed in base branch"
-          }
-        },
-        commit: {
-          message: "MarkdownLint fixes"
-        }
-    }'
-    )
-echo $push > /atm/output/push.json
+# print message to stderr prefixed by package name.
+# usage: err MESSAGE
+function err () {
+    msg "$*" 1>&2
+}
 
-##### Prepare command arguments
-if ! [[ -z "$fix" ]]; then
-  fix_option="--fix"
-fi
+function main () {
+    # Extract some skill configuration from the incoming event payload
+    local fix config ignores labels branch
+    fix=$( < "$ATOMIST_PAYLOAD" \
+               jq -r '.skill.configuration.instances[0].parameters[] | select( .name == "push_strategy" ) | .value' )
+    config=$( < "$ATOMIST_PAYLOAD" \
+                  jq -r '.skill.configuration.instances[0].parameters[] | select( .name == "config" ) | .value' )
```

Comment:
Should this use `-r` if the config value is JSON? I guess it does not matter as `-r` really only affects strings.